packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,184 @@
+import { type Notification } from '../../../lib/notifications/models/notification.model'
+import DirectShareApi, { PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE } from '../../../lib/share/direct-share.api'
+import { patient1Id } from '../data/patient.api.data'
+
+export const checkCaregiversListLayout = async () => {
+  expect(await screen.findByTestId('patient-caregivers-list')).toBeVisible()
+
+  const addCaregiverButton = screen.getByRole('button', { name: 'Add caregiver' })
+  expect(addCaregiverButton).toBeVisible()
+  expect(addCaregiverButton).toBeEnabled()
+
+  const caregiversTable = screen.getByTestId('patient-caregivers-list')
+  expect(caregiversTable).toHaveTextContent('Last nameFirst nameEmail')
+
+  const caregiverRow = within(caregiversTable).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}`)
+  expect(caregiverRow).toBeVisible()
+  expect(caregiverRow).toHaveTextContent('UserCaregivercaregiver@mail.com')
+  expect(within(caregiverRow).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}-button-remove`)).toBeVisible()
+}
+
+export const checkAddCaregiverSuccess = async (newCaregiverEmail: string) => {
+  const addCaregiverButton = screen.getByRole('button', { name: 'Add caregiver' })
+  await userEvent.click(addCaregiverButton)
+
+  const addCaregiverDialog = screen.getByRole('dialog')
+  expect(addCaregiverDialog).toBeVisible()
+  expect(within(addCaregiverDialog).getByText('Add a caregiver')).toBeVisible()
+  const emailInput = within(addCaregiverDialog).getByRole('textbox', { name: 'Email' })
+  expect(emailInput).toBeVisible()
+
+  await userEvent.type(emailInput, newCaregiverEmail)
+  const inviteButton = within(addCaregiverDialog).getByRole('button', { name: 'Invite' })
+  expect(inviteButton).toBeEnabled()
+
+  jest.spyOn(NotificationApi, 'getSentInvitations').mockResolvedValueOnce([{
+    email: newCaregiverEmail,
+    type: NotificationType.directInvitation,
+    target: { id: 'target-id' },
+    id: 'my-id'
+  } as Notification])
+  await userEvent.click(inviteButton)
+
+  expect(DirectShareApi.addDirectShare).toHaveBeenCalledWith(patient1Id, newCaregiverEmail)
+  expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+
+  const inviteSuccessfulSnackbar = screen.getByTestId('alert-snackbar')
+  expect(inviteSuccessfulSnackbar).toHaveTextContent('Invite sent!')
+  await userEvent.click(within(inviteSuccessfulSnackbar).getByTitle('Close'))
+}
+
+export const checkAddCaregiverCancel = async () => {
+  const addCaregiverButton = screen.getByRole('button', { name: 'Add caregiver' })
+  await userEvent.click(addCaregiverButton)
+
+  const addCaregiverDialog = screen.getByRole('dialog')
+  expect(addCaregiverDialog).toBeVisible()
+
+  const cancelButton = within(addCaregiverDialog).getByRole('button', { name: 'Cancel' })
+  expect(cancelButton).toBeVisible()
+  expect(cancelButton).toBeEnabled()
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏These two checks are useless as you would not be able to click on the button if these conditions were not met :slightly_smiling_face: 

```suggestion
```